### PR TITLE
Move to Android 11

### PR DESCRIPTION
**Pull request by grote**
After this PR has been merged into `master`, it will not build anymore within AOSP 10. People wanting to maintain Seedvault for Android 10 are welcome to submit PRs for the [`android10`](https://github.com/stevesoltys/seedvault/commits/android10) branch.

I've done all my testing and dev work on top of this PR for the past weeks. It is included in GrapheneOS 11 preview release. Therefore, I consider it stable and tested enough to be merged into `master`. After additional testing and confidence, we can branch off a stable `android11` branch from it.

### CODE REVIEW

**Comment by chirayudesai** [comment]
I will push the current android10 to a 'master-android10' in my fork so that we can PR it for merge to android10 later.

I want to get translations merged too for android10, but that's not a blocker as they can be picked up separately without issue.

**Comment by chirayudesai** [comment]
> I will push the current android10 to a 'master-android10' in my fork so that we can PR it for merge to android10 later.

Done, see #135 
> 
> I want to get translations merged too for android10, but that's not a blocker as they can be picked up separately without issue.

Done, #134 


This means we can merge this freely into master, and still have something to merge into android10 once tested.

File: app/src/test/java/com/stevesoltys/seedvault/metadata/MetadataManagerTest.kt [hunk 1]
```diff
@@ -34,6 +35,7 @@ import java.io.IOException
 import kotlin.random.Random
 
 @RunWith(AndroidJUnit4::class)
+@Config(sdk = [29]) // robolectric does not support 30, yet
```
Comment (chirayudesai) [inline app/src/test/java/com/stevesoltys/seedvault/metadata/MetadataManagerTest.kt:39]: Do we know when are they going to?

I think maybe there was a link to some upstream issue - would be nice to have that somewhere, can even be a separate issue here "Update roboelectric once upstream is fixed" or something.

Comment (grote) [reply app/src/test/java/com/stevesoltys/seedvault/metadata/MetadataManagerTest.kt:39]: No idea. They've just got around to support 29 properly.

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt [hunk 2]
```diff
@@ -19,8 +19,6 @@ class RestoreActivity : RequireProvisioningActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        if (isSetupWizard) hideSystemUI()
```
Comment (chirayudesai) [inline app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: Not needed anymore?

File: Android.bp [hunk 3]
```diff
@@ -44,7 +44,10 @@ android_app {
     required: [
         "privapp_whitelist_com.stevesoltys.backup",
         "com.stevesoltys.backup_whitelist"
-    ]
+    ],
+    optimize: {
+        enabled: false,
```
Comment (chirayudesai) [inline Android.bp:49]: `java.lang.NoClassDefFoundError` was the issue right?

Could setup a custom `proguard.flags` in the future if needed.

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt (hunk 2 above)
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: I think you said that the system UI should still be shown in SUW, right? The code for doing that changed on Android 11.

**Review by chirayudesai — COMMENTED** [review]
Did you try this with SetupWizard yet?

Asking due to the removal of the `SYSTEM_UI_FLAG_HIDE_NAVIGATION` bit.

LGTM.

File: Android.bp (hunk 3 above)
Comment (grote) [reply Android.bp:49]: That actually had another reason, so we *could* re-enable proguard, but it is risky.

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt (hunk 2 above)
Comment (chirayudesai) [reply app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: But I don't see any new code that still does this. Should make a note somewhere if this needs updating.

File: Android.bp (hunk 3 above)
Comment (chirayudesai) [reply Android.bp:49]: Ah. I don't mind just leaving it disabled, definitely the better option for now.

**Review by chirayudesai — APPROVED** [review]

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt (hunk 2 above)
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: Yes due to #63 I didn't add the new code.

Comment (chirayudesai) [reply app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: Ah, I see the confusion, that shouldn't say *and* - we want it to look like what it currently does / has for the year so far.

**Review by chirayudesai — CHANGES_REQUESTED** [review]
We just need to bring back https://github.com/stevesoltys/seedvault/commit/65905a7160cbae595aba3f05810532931042bbfb to undo the changes in the "Android 11 warnings" commit.

Can leave a note in the general Android 11 to update this to a newer API if available.

File: app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt (hunk 2 above)
Comment (grote) [reply app/src/main/java/com/stevesoltys/seedvault/restore/RestoreActivity.kt:22]: alright, I added it back.

**Comment by chirayudesai** [comment]
Can you amend "Bring back hiding of navigation bar in SetupWizard" into "Fix compile warnings on AOSP 11"

Just makes it confusing otherwise when looking at history.

Untested personally, but code looks good and good to go otherwise.

**Review by chirayudesai — APPROVED** [review]
857f503a8fcd511136227ec0b38426a0c491d7f8 - the latest, is indentical to the commit before amending the setupwizard change - i.e. a7dc7f52d340243e254f129df958624329985af1

Thanks!

Edit: Identical as in `git diff` output is empty, the only change is that the new extra commit is gone and amended into "Fix compile warnings in AOSP 11"

**Pull request merged**